osu.Game/Graphics/UserInterface/SettingsDropdownContainer.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics.Sprites;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public abstract class SettingsDropdownContainer : Container
+    {
+        /// <summary>
+        /// The title of this container, which will be written in header.
+        /// </summary>
+        protected abstract string Title { get; }
+
+        private const float transition_duration = 600;
+        private const int container_width = 270;
+        private const int border_thickness = 2;
+        private const int header_height = 30;
+        private const int corner_radius = 5;
+
+        private FillFlowContainer content;
+        private SimpleButton button;
+        private bool buttonIsActive;
+
+        private Color4 buttonActiveColour;
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours)
+        {

```

Comment:
All of this method should be moved to the constructor, after which `content` and `button` can be made readonly. Use load() for initializing the button colour.

Generally I think we should keep ctors for object initialization and load() for virtual methods + DI initialization. Having this stuff inside load() messes with object initializers because the load() method would override them - possibly unexpectedly.